# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/arcticfox/hotfix-cpp.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Since 47.0 fails with python exception:
OSError: [Errno 13]  Permission denied

added in:
https://github.com/rmottola/Arctic-Fox/commit/6abc39adccdd9447f8b6a76745664bbd8980f56d

--- Arctic-Fox-47.0/dom/bindings/GenerateCSS2PropertiesWebIDL.py.vanilla	2026-02-14 21:29:03.711514445 +0100
+++ Arctic-Fox-47.0/dom/bindings/GenerateCSS2PropertiesWebIDL.py	2026-02-14 21:29:11.287000000 +0100
@@ -16,7 +16,7 @@
     return "  [%s] attribute DOMString %s;\n" % (", ".join(extendedAttrs),
                                                  propName)
 def generate(output, idlFilename, preprocessorHeader):
-    cpp = list(buildconfig.substs['CPP'])
+    cpp = shellutil.split(buildconfig.substs['CPP'])
     cpp += shellutil.split(buildconfig.substs['ACDEFINES'])
     cpp.append(preprocessorHeader)
     preprocessed = subprocess.check_output(cpp)
--- Arctic-Fox-47.0/layout/style/GenerateCSSPropsGenerated.py.vanilla	2026-02-14 21:28:21.420000000 +0100
+++ Arctic-Fox-47.0/layout/style/GenerateCSSPropsGenerated.py	2026-02-14 21:28:43.420548020 +0100
@@ -10,7 +10,7 @@
 from mozbuild import shellutil
 
 def get_properties(preprocessorHeader):
-    cpp = list(buildconfig.substs['CPP'])
+    cpp = shellutil.split(buildconfig.substs['CPP'])
     cpp += shellutil.split(buildconfig.substs['ACDEFINES'])
     cpp.append(preprocessorHeader)
     preprocessed = subprocess.check_output(cpp)
